app.ts

```diff
@@ -0,0 +1,292 @@
+function getTable(): Promise<void> {
+	const content: HTMLElement | null = document.getElementById("content");
+	let toParameter = getParameters(fromParameter);
+	const pageStats: HTMLElement | null = document.getElementById("pageStats");
+
+	// Clears Table
+	if (content !== null) {
+		content.innerHTML = "";
+	}
+
+	// Displays The Current Results Being Shown
+	let currentStats = `Showing results from ${fromParameter} to ${toParameter} out of ${recordCount} results.`;
+	if (pageStats !== null) {
+		pageStats.innerHTML = currentStats;
+	}
+
+	return fetch(`http://localhost:2050/records?from=${fromParameter}&to=${toParameter}`, {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((data: string) => {
+			let contentData = data;
+			for (let content of contentData) {
+				createTableContent(content);
+			}
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+// Gets Total Of All Records
+function getRecordCount(): Promise<void> {
+	return fetch("http://localhost:2050/recordCount", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((data: number) => {
+			recordCount = data;
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+//// Debounce
+function debounce(fn: any, delay: number) {
+	return function () {
+		clearTimeout(timer);
+		timer = setTimeout(() => {
+			fn();
+		}, delay);
+	};
+}
+
+//// Sizing And Resizing
+function resizing() {
+	let end = fromParameter + getNoOfRows();
+	let toParameter: number;
+	let maxRecordsID = recordCount - 1;
+
+	if (end > maxRecordsID) {
+		toParameter = maxRecordsID;
+		fromParameter = toParameter - getNoOfRows();
+	}
+	getTable();
+}
+
+//// Navigation
+// Next
+class Next {
+	nextCount: number;
+	nextButton: HTMLElement | null;
+
+	constructor() {
+		this.nextCount = 0;
+		this.nextButton = document.getElementById("next");
+
+		const nextDebounce = (fn: any, delay: number) => {
+			return () => {
+				this.nextCount++;
+				clearTimeout(timer);
+				timer = setTimeout(() => {
+					fn();
+				}, delay);
+			};
+		};
+
+		let next = () => {
+			let toParameter = getParameters(fromParameter);
+			let maxRecordsID = recordCount - 1;
+
+			if (toParameter === maxRecordsID) {
+				alert("You have reached the final page");
+			}
+
+			let nextAmount = toParameter - fromParameter + 1;
+			let nextCountAmount = nextAmount * this.nextCount;
+			fromParameter = fromParameter + nextCountAmount;
+			toParameter = fromParameter + getNoOfRows();
+
+			let end = fromParameter + getNoOfRows();
+
+			if (end > maxRecordsID) {
+				toParameter = maxRecordsID;
+				fromParameter = toParameter - getNoOfRows();
+			}
+
+			this.nextCount = 0;
+
+			getTable();
+		};
+
+		if (this.nextButton) {
+			this.nextButton.addEventListener("click", nextDebounce(next, 500));
```

Comment:
If you are dead set on debouncing this button, then you should rather just call the `next` function here, and put your debounce inside the `next` function on the `getTable` function. Then you can get rid of the `nextCount` logic